src/object-store/src/results.cpp

```diff
@@ -326,8 +408,21 @@ Query Results::get_query() const
         case Mode::Empty:
         case Mode::Query:
             return m_query;
-        case Mode::TableView:
-            return m_table_view.get_query();
+        case Mode::TableView: {
+            // A TableView has an associated Query if it was produced by Query::find_all. This is indicated
+            // by TableView::get_query returning a Query with a non-null table.
+            Query query = m_table_view.get_query();
+            if (query.get_table()) {
+                return query;
+            }
+
+            // The TableView has no associated query so create one with no conditions that is restricted
+            // to the rows in the TableView.
+            m_table_view.sync_if_needed();
+            return Query(*m_table, (std::unique_ptr<TableViewBase>)std::make_unique<TableView>(m_table_view));
```

Comment:
You can just do `make_unique<TableViewBase>` rather than use a C-style cast, right?